libsolidity/formal/SMTChecker.cpp

```diff
@@ -1073,35 +1076,38 @@ smt::Expression SMTChecker::arithmeticOperation(
 		_op == Token::Add ? _left + _right :
 		_op == Token::Sub ? _left - _right :
 		_op == Token::Div ? division(_left, _right, intType) :
-		/*op == Token::Mul*/ _left * _right
+		_op == Token::Mul ? _left * _right :
+		/*_op == Token::Mod*/ _left % _right
 	);
 
-	if (_op == Token::Div)
+	if (_op == Token::Div || _op == Token::Mod)
 	{
 		checkCondition(_right == 0, _location, "Division by zero", "<result>", &_right);
 		m_interface->addAssertion(_right != 0);
 	}
-
-	addOverflowTarget(
```

Comment:
What about the ``(-2^255) / -1`` overflow? Not sure that was handled correctly before (haven't seen tests for it) and not sure it needs to be addressed here, but does that work correctly or is there an issue for it?

Reply:
Fixed the code and added boost test